### PR TITLE
gh-95205: Improve WASM README.md (GH-95267)

### PR DESCRIPTION
**Pull request by tiran**
- drop ``emrun``
- add build and debug instructions for WASI
- add wasi-env helper script

* Issue: gh-95205

### CODE REVIEW

File: Tools/wasm/README.md [hunk 1]
```diff
@@ -69,12 +69,9 @@ popd
...
 Serve `python.html` with a local webserver and open the file in a browser.
-
-```shell
-emrun builddir/emscripten-browser/python.html
-```
-
-or
+Python comes with a minimal web server script that sets necessary HTTP
+headers like COOP, COEP, and mimetypes. Run the script outside the container
+and from the root of the CPython checkout
```
Comment (erlend-aasland) [inline Tools/wasm/README.md]:
```suggestion
and from the root of the CPython checkout.
```

File: Tools/wasm/README.md [hunk 2]
```diff
@@ -232,6 +230,28 @@ WASI builds require [WASI SDK](https://github.com/WebAssembly/wasi-sdk) 15.0+
 and currently [wasix](https://github.com/singlestore-labs/wasix) for POSIX
 compatibility stubs.
 
+## Cross-compile to wasm32-wasi
+
+The script ``wasi-env`` sets necessary compiler and linker flags as well as
+``pkg-config`` overrides. The script assumes that WASI-SDK is installed in
+``/opt/wasi-sdk`` or ``$WASI_SDK_PATH``.
```
Comment (erlend-aasland) [inline Tools/wasm/README.md:237]: Markdown, not reST ;)

```suggestion
The script `wasi-env` sets necessary compiler and linker flags as well as
`pkg-config` overrides. The script assumes that WASI-SDK is installed in
`/opt/wasi-sdk` or `$WASI_SDK_PATH`.
```

File: Tools/wasm/README.md [hunk 3]
```diff
@@ -376,6 +396,16 @@ git clone https://github.com/emscripten-core/emsdk.git /opt/emsdk
 /opt/emsdk/emsdk activate latest
 ```
 
+### Optionally: enable ccache for EMSDK
+
+The ``EM_COMPILER_WRAPPER`` must be set after the EMSDK environment is
```
Comment (erlend-aasland) [inline Tools/wasm/README.md:401]:
```suggestion
The `EM_COMPILER_WRAPPER` must be set after the EMSDK environment is
```

File: Tools/wasm/README.md [hunk 4]
```diff
@@ -424,3 +450,9 @@ ln -srf -t /usr/local/bin/ ~/.wasmtime/bin/wasmtime
 git clone https://github.com/singlestore-labs/wasix.git ~/wasix
 make install -C ~/wasix
 ```
+
+### WASI debugging
+
+* ``wasmtime run -g`` generates debugging symbols for gdb and lldb.
+* The environment variable ``RUST_LOG=wasi_common`` enables debug and
```
Comment (erlend-aasland) [inline Tools/wasm/README.md:457]:
```suggestion
* `wasmtime run -g` generates debugging symbols for gdb and lldb.
* The environment variable `RUST_LOG=wasi_common` enables debug and
```

**Review by erlend-aasland — APPROVED** [review]

**Comment by tiran** [comment]
I have been using double ticks for code in MD for years. They work fine.

**Comment by tiran** [comment]
I'm keeping the double backticks because a) muscle memory, b) consistency with rest of the document, c) they work. :)

**Pull request merged**